crates/symbolicator-proguard/src/symbolication.rs

```diff
@@ -184,17 +184,30 @@ impl ProguardService {
         frame: &JvmFrame,
         release_package: Option<&str>,
     ) -> Vec<JvmFrame> {
+        let deobfuscated_signature: Option<proguard::DeobfuscatedSignature> =
+            frame.signature.as_ref().and_then(|signature| {
+                for mapper in mappers {
+                    if let Some(deobfuscated_signature) = mapper.deobfuscate_signature(signature) {
+                        return Some(deobfuscated_signature);
+                    }
+                }
+                None
+            });
+        let mut params = deobfuscated_signature.as_ref().map(|sig| {
+            String::from_iter(sig.parameters_types().map(|param| format!("{},", param)))
+        });
         let stack_frame = frame
             .lineno
             .map(|lineno| {
                 proguard::StackFrame::new(&frame.module, &frame.function, lineno as usize)
             })
             .or_else(|| {
-                frame.parameters.as_ref().map(|params| {
+                params.as_mut().map(|p| {
+                    p.pop();
```

Comment:
It took me quite some time to figure out that you are popping off the trailing `,` here. Please do that in the initial statement that creates `params`. It doesn’t need to be `mut` so you can mutate it 10 lines after creating it, where its very hard to follow whats actually happening.

Reply:
@Swatinem I've actually removed that already but I haven't pushed yet.

I'm cleaning the code. I'll ask for a new review once I'm done.